modules/heat_conduction/src/fvbcs/FVThermalResistanceBC.C

```diff
@@ -0,0 +1,173 @@
+  return params;
+}
+
+FVThermalResistanceBC::FVThermalResistanceBC(const InputParameters & parameters)
+  : FVFluxBC(parameters),
+    _geometry(getParam<MooseEnum>("geometry").getEnum<Moose::CoordinateSystemType>()),
+    _inner_radius(
+        _geometry == Moose::CoordinateSystemType::COORD_RZ ? getParam<Real>("inner_radius") : 1.0),
+    _T_ambient(getParam<Real>(HeatConduction::T_ambient)),
+    _k(getParam<std::vector<Real>>("thermal_conductivities")),
+    _dx(getParam<std::vector<Real>>("conduction_thicknesses")),
+    _h(getADMaterialPropertyByName<Real>(getParam<MaterialPropertyName>("htc"))),
+    _emissivity(getParam<Real>(HeatConduction::emissivity)),
+    _max_iterations(getParam<unsigned int>("max_iterations")),
+    _tolerance(getParam<Real>("tolerance")),
+    _alpha(getParam<Real>("step_size")),
+    _T_surface(0.0),
+    _outer_radius(_inner_radius),
+    _conduction_resistance(0.0),
+    _parallel_resistance(0.0)
+{
+  if (_k.size() != _dx.size())
+    mooseError("Number of specified thermal conductivities must match the number "
+               "of conduction layers!");
+
+  if (_geometry == Moose::CoordinateSystemType::COORD_RZ)
+    for (const auto & d : _dx)
+      _outer_radius += d;
+
+  // because the thermal conductivities are constant, we only need to compute
+  // the conduction resistance one time
+  computeConductionResistance();
+}
+
+void
+FVThermalResistanceBC::computeConductionResistance()
+{
+  ADReal r = _inner_radius;
+
+  for (std::size_t i = 0; i < _k.size(); ++i)
+  {
+    switch (_geometry)
+    {
+      case Moose::CoordinateSystemType::COORD_XYZ:
+        _conduction_resistance += _dx[i] / _k[i];
+        break;
+      case Moose::CoordinateSystemType::COORD_RZ:
+      {
+        _conduction_resistance += std::log((_dx[i] + r) / r) / _k[i];
+        r += _dx[i];
+        break;
+      }
+      default:
+        mooseError("Unhandled 'GeometryEnum' in 'FVThermalResistanceBC'!");
+    }
+  }
+}
+
+ADReal
+FVThermalResistanceBC::computeQpResidual()
+{
+  // radiation resistance has to be solved iteratively, since we don't know the
+  // surface temperature. We do know that the heat flux in the conduction layers
+  // must match the heat flux in the parallel convection-radiation segment. For a
+  // first guess, take the surface temperature as the average of _u and T_ambient.
+  _T_surface = 0.5 * (_u[_qp] + _T_ambient);
+
+  // total flux perpendicular to boundary
+  ADReal flux;
+
+  // resistance component representing the sum of the convection and radiation
+  // resistances, in parallel
+  computeParallelResistance();
+
+  // other iteration requirements
+  Real iteration = 0;
+  ADReal norm = 2 * _tolerance;
+  ADReal T_surface_previous;
+
+  // iterate to find the approximate surface temperature needed for evaluating the
+  // radiation resistance. We only do this iteration if we have radiation transfer.
+  if (_emissivity > HeatConduction::DefaultValues::epsilon)
+    while (norm > _tolerance)
```

Comment:
I'd multiply tolerance by the relaxation factor to avoid reaching convergence just because of strong relaxation